tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/GlobalPagingTest.java

```diff
@@ -334,7 +334,7 @@ public void testManagementMessageRequestCannotFailAfterFailedDirectDeliver() thr
                   final MessageReference reference = MessageReference.Factory.createReference(session.createMessage(false), queue);
                   //it will cause QueueImpl::directDeliver -> false
                   queue.addHead(reference, false);
-                  Assert.assertSame(reference, queue.removeReferenceWithID(reference.getMessageID()));
+                  queue.removeReferenceWithID(reference.getMessageID());
```

Comment:
Can you assertNotNull instead instead of just ignore the result?

Reply:
But it can be null, is just mater of timing: the remove itself is not necessary to the test TBH (I've written that test)

Reply:
@franz1981 make it not racy then:

```java
Wait.assertNotNull(() -> queue.removReferenceWithID(reference.getMessageID())
```


It seems to me you could even use Wait.assertSame